src/lib.rs

```diff
@@ -571,6 +571,37 @@ impl CecLogicalAddresses {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum TryFromCecLogicalAddressesError {
+    UnknownPrimaryAddress,
+    InvalidPrimaryAddress,
+}
+
+impl TryFrom<cec_logical_addresses> for CecLogicalAddresses {
+    type Error = TryFromCecLogicalAddressesError;
+    fn try_from(addresses: cec_logical_addresses) -> Result<Self, Self::Error> {
+        let primary = CecLogicalAddress::try_from(addresses.primary)
+            .map_err(|_| TryFromCecLogicalAddressesError::UnknownPrimaryAddress)?;
```

Comment:
If `CecLogicalAddress` construction fails, should we instead raise `TryFromCecLogicalAddressesError::InvalidPrimaryAddress` error here?

Reply:
Fixed